### PR TITLE
[SPARK-47593][CORE] Connector module: Migrate logWarn with variables to structured logging framework

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to migrate `logWarning` in module `Connector` with variables to `structured logging framework`.

### Why are the changes needed?
To enhance Apache Spark's logging system by implementing structured logging.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
- Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/GrpcRetryHandler.scala [hunk 1]
```diff
@@ -197,18 +200,20 @@ private[sql] object GrpcRetryHandler extends Logging {
 
         if (time.isDefined) {
           logWarning(
-            s"Non-Fatal error during RPC execution: $lastException, retrying " +
-              s"(wait=${time.get.toMillis}, currentRetryNum=$currentRetryNum, " +
-              s"policy: ${policy.getName})")
+            log"Retrying (wait=${MDC(WAIT_TIME, time.get.toMillis)}, " +
```
Comment (gengliangwang) [inline connector/connect/common/src/main/scala/org/apache/spark/sql/connect/client/GrpcRetryHandler.scala:203]: We can also use `ERROR` for lastException. Either way is ok.

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala [hunk 2]
```diff
@@ -268,9 +269,11 @@ private[connect] class ExecuteGrpcResponseSender[T <: Message](
       if (interrupted) {
         // This sender got interrupted. Kill this RPC.
         logWarning(
-          s"Got detached from opId=${executeHolder.operationId} at index ${nextIndex - 1}." +
-            s"totalTime=${System.nanoTime - startTime}ns " +
-            s"waitingForResults=${consumeSleep}ns waitingForSend=${sendSleep}ns")
+          log"Got detached from opId=${MDC(OP_ID, executeHolder.operationId)} " +
+            log"at index ${MDC(INDEX, nextIndex - 1)}." +
+            log"totalTime=${MDC(TOTAL_TIME, System.nanoTime - startTime)}ns " +
```
Comment (gengliangwang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala:274]: Please use ms

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala [hunk 3]
```diff
@@ -268,9 +269,11 @@ private[connect] class ExecuteGrpcResponseSender[T <: Message](
...
+            log"waitingForResults=${MDC(WAIT_RESULT_TIME, consumeSleep)}ns " +
```
Comment (gengliangwang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala:275]: Please use ms

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala [hunk 4]
```diff
@@ -268,9 +269,11 @@ private[connect] class ExecuteGrpcResponseSender[T <: Message](
...
+            log"waitingForSend=${MDC(WAIT_SEND_TIME, sendSleep)}ns")
```
Comment (gengliangwang) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteGrpcResponseSender.scala:276]: Please use ms

File: connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala [hunk 5]
```diff
@@ -332,7 +333,7 @@ class KafkaTestUtils(
         Utils.deleteRecursively(new File(f))
       } catch {
         case e: IOException if Utils.isWindows =>
-          logWarning(e.getMessage)
+          logWarning(log"${MDC(ERROR, e.getMessage)}", e)
```
Comment (gengliangwang) [inline connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala:336]:
```suggestion
          logWarning(log"${MDC(ERROR, e.getMessage)}")
```

File: connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala [hunk 6]
```diff
@@ -653,13 +654,13 @@ class KafkaTestUtils(
         Utils.deleteRecursively(snapshotDir)
       } catch {
         case e: IOException if Utils.isWindows =>
-          logWarning(e.getMessage)
+          logWarning(log"${MDC(ERROR, e.getMessage)}", e)
```
Comment (gengliangwang) [inline connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala:657]: ditto

File: connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala [hunk 7]
```diff
@@ -653,13 +654,13 @@ class KafkaTestUtils(
...
       }
       try {
         Utils.deleteRecursively(logDir)
       } catch {
         case e: IOException if Utils.isWindows =>
-          logWarning(e.getMessage)
+          logWarning(log"${MDC(ERROR, e.getMessage)}", e)
```
Comment (gengliangwang) [inline connector/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala:663]: ditto

**Comment by gengliangwang** [comment]
@panbingkun Thanks for the works! LGTM except for some minor comments.

**Comment by panbingkun** [comment]
> @panbingkun Thanks for the works! LGTM except for some minor comments.

@gengliangwang all done. Thanks!

**Review by gengliangwang — APPROVED** [review]

**Comment by gengliangwang** [comment]
Thanks, merging to master

**Pull request closed without merging**